solnlib/conf_manager.py

```diff
@@ -557,3 +550,68 @@ def get_log_level(
             f"taking {default_log_level} as log level."
         )
         return default_log_level
+
+
+def get_proxy_dict(
+    logger: logging.Logger,
+    session_key: str,
+    app_name: str,
+    conf_name: str,
+    proxy_stanza: str = "proxy",
+    **kwargs,
+) -> Union[Dict[str, str], NoReturn]:
+    """This function returns the log level for the addon from configuration
+    file.
+
+    Arguments:
+        logger: Logger.
+        session_key: Splunk access token.
+        app_name: Add-on name.
+        conf_name: Configuration file name where logging stanza is.
+        proxy_stanza: Proxy stanza that would contain the Proxy details
+    Returns:
+        A dictionary is returned with stanza details present in the file.
+        The keys related to `eai` are removed before returning.
+
+    Examples:
+        >>> from solnlib import conf_manager
+        >>> proxy_details = conf_manager.get_proxy_dict(
+        >>>     logger,
+        >>>     "session_key",
+        >>>     "ADDON_NAME",
+        >>>     "splunk_ta_addon_settings",
+        >>> )
+    """
+    proxy_dict = {}
+    try:
+        cfm = ConfManager(
+            session_key,
+            app_name,
+            realm=f"__REST_CREDENTIAL__#{app_name}#configs/conf-{conf_name}",
+        )
+        conf = cfm.get_conf(conf_name)
+    except ConfManagerException:
+        logger.error(f"Failed to fetch configuration file '{conf_name}'.")
+    else:
+        try:
+            proxy_dict = conf.get(proxy_stanza)
+        except ConfStanzaNotExistException:
+            logger.error(
+                f"Failed to fetch '{proxy_stanza}' from the configuration file '{conf_name}'. "
+            )
+        else:
+            # remove the other fields that are added by ConfFile class
+            proxy_dict.pop("disabled", None)
+            proxy_dict.pop("eai:access", None)
+            proxy_dict.pop("eai:appName", None)
+            proxy_dict.pop("eai:userName", None)
```

Comment:
are those fields always added? If we try to pop non-existing key we will get KeyError, maybe worth to handle this?